### PR TITLE
[air/docs] Install required dependencies in docstrings test, convert XGBoost predictor test

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

Some of our docstrings are outdated: They may use stale APIs, deprecated features, or plainly don't work. To make sure our documentation is up to date, we should test docstrings more rigorously. This requires us to move from RST code blocks to doctest strings.
This PR adjusts our docstring test to install required dependencies and includes an example conversion of an xgboost test.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.build.yml [hunk 1]
```diff
@@ -539,16 +539,11 @@
...
     - export LINT=1
-    - echo "--- Setting up Python 3.7 environment."
-    - ./ci/env/install-dependencies.sh
-    # Specifying above somehow messes up the Ray install.
-    # Uninstall and re-install Ray so that we can use Ray Client
-    # (remove thirdparty_files to sidestep an issue with psutil).
-    - pip uninstall -y ray && rm -rf /ray/python/ray/thirdparty_files
-    - pushd /ray && git clean -f -f -x -d -e .whl -e python/ray/dashboard/client && popd
+    - DOC_TESTING=1 INSTALL_HOROVOD=1 DATA_PROCESSING_TESTING=1 TRAIN_TESTING=1 TUNE_TESTING=1 RLLIB_TESTING=1 PYTHON=3.7 ./ci/env/install-dependencies.sh
     - bazel clean --expunge
```
Comment (matthewdeng) [inline .buildkite/pipeline.build.yml:547]: A little orthogonal, but can you add the step to print out dependencies here?

```
./ci/env/env_info.sh
```

File: python/ray/train/xgboost/xgboost_trainer.py [hunk 2]
```diff
@@ -28,22 +28,22 @@ class XGBoostTrainer(GBDTTrainer):
...
+    >>> train_dataset = ray.data.from_items(
+    ...     [{"x": x, "y": x + 1} for x in range(32)])
+    >>> trainer = XGBoostTrainer(
+    ...     label_column="y",
+    ...     params={"objective": "reg:squarederror"},
+    ...     scaling_config=ScalingConfig(num_workers=3),
+    ...     datasets={"train": train_dataset}
+    ... )
+    >>> result = trainer.fit() # doctest: +ELLIPSIS
+    == Status ...
```
Comment (matthewdeng) [inline python/ray/train/xgboost/xgboost_trainer.py:46]: nit: I'm personally not familiar with doctesting, so this seemed kind of weird to me. Should we have consistency with  `stopper.py`? https://github.com/ray-project/ray/blob/e7d9c242afab6028a2dd87def73389969d1dbd70/python/ray/tune/stopper/stopper.py#L36-L37

**Review by matthewdeng — APPROVED** [review]

File: .buildkite/pipeline.build.yml (hunk 1 above)
Comment (krfricke) [reply .buildkite/pipeline.build.yml:547]: Good idea, added

File: python/ray/train/xgboost/xgboost_trainer.py (hunk 2 above)
Comment (krfricke) [reply python/ray/train/xgboost/xgboost_trainer.py:46]: Hm last time this didn't pass for me but wemay have upgraded doctest in the meantime or so. I'll adjust and check

**Comment by krfricke** [comment]
This is being taken over by someone else

**Pull request closed without merging**